File: ios/View/SmileIDDocumentCapture.swift
import Foundation
import SmileID
import SwiftUI

struct SmileIDDocumentCapture: View {
    let showInstructions: Bool
    let showAttribution: Bool
    let allowGallerySelection: Bool
    let showSkipButton: Bool
    let instructionsHeroImage: UIImage
    let instructionsTitleText: String
    let instructionsSubtitleText: String
    let captureTitleText: String
    let knownIdAspectRatio: Double?
    let showConfirmation: Bool
    @ObservedObject var viewModel: SelfieViewModel
    var body: some View {
        NavigationView {
            DocumentCaptureScreen(
                showInstructions: self.showInstructions,
                showAttribution: self.showAttribution,
                allowGallerySelection: self.allowGallerySelection,
                showSkipButton: self.showSkipButton,
                instructionsHeroImage: self.instructionsHeroImage,
                instructionsTitleText: self.instructionsTitleText,
                instructionsSubtitleText: self.instructionsSubtitleText,
                captureTitleText: self.captureTitleText,
                knownIdAspectRatio: self.knownIdAspectRatio,
                showConfirmation: self.showConfirmation,
                onConfirm :{ data in
                    
                },
                onError: { error in
                    
                },
                onSkip: {
                    
                })
        }.navigationViewStyle(StackNavigationViewStyle())
    }
}

File: android/src/main/java/com/smileidentity/react/views/SmileIDDocumentCapture.kt
package com.smileidentity.react.views

import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.ui.res.stringResource
import androidx.lifecycle.viewmodel.compose.LocalViewModelStoreOwner
import com.facebook.react.bridge.ReactApplicationContext
import com.smileidentity.R
import com.smileidentity.compose.components.LocalMetadata
import com.smileidentity.compose.document.DocumentCaptureScreen
import com.smileidentity.compose.document.DocumentCaptureSide
import com.smileidentity.util.randomJobId
import com.smileidentity.util.randomUserId

class SmileIDDocumentCapture(context: ReactApplicationContext) : SmileIDView(context) {
  var countryCode: String? = null
  var allowGalleryUpload: Boolean? = false
  var idAspectRatio: Float? = 1.0f
  override fun renderContent() {
    composeView.apply {
      val customViewModelStoreOwner = CustomViewModelStoreOwner()
      setContent {
        CompositionLocalProvider(LocalViewModelStoreOwner provides customViewModelStoreOwner) {
          val userId = userId ?: rememberSaveable { randomUserId() }
          val jobId = jobId ?: rememberSaveable { randomJobId() }
          val metadata = LocalMetadata.current
          DocumentCaptureScreen(
            jobId = jobId,
            side = DocumentCaptureSide.Front,
            showInstructions = showInstructions ?: true,
            showAttribution = showAttribution ?: true,
            allowGallerySelection = allowGalleryUpload ?: false,
            showSkipButton = false,
            instructionsHeroImage = R.drawable.si_doc_v_front_hero,
            instructionsTitleText = stringResource(R.string.si_doc_v_instruction_title),
            instructionsSubtitleText = stringResource(
              id = R.string.si_verify_identity_instruction_subtitle,
            ),
            captureTitleText = stringResource(
              id = R.string.si_doc_v_capture_instructions_front_title,
            ),
            knownIdAspectRatio = idAspectRatio,
            onConfirm = { _ ->

            },
            onError = {throwable: Throwable ->
                emitFailure(throwable)
            },
          )
        }
      }
    }
  }
}
